app/Filament/Resources/NewsResource.php

```diff
@@ -45,6 +50,26 @@ public static function form(Form $form): Form
                             ->label('URL')
                             ->required()
                             ->activeUrl(),
+
+                        Forms\Components\Toggle::make('pinned_at')
+                            ->label('Pinned')
+                            ->helperText('If enabled, this will be sorted to the top of the news until unpinned.')
+                            ->columnSpanFull()
+                            ->disabled(fn (?News $record) => !$record || !$user->can('pin', $record))
+                            ->dehydrated()
+                            ->afterStateHydrated(function (?News $record, $component) {
+                                if (!$record) {
+                                    $component->state(false);
+
+                                    return;
+                                }
+
+                                $component->state(!is_null($record->pinned_at));
+                            })
+                            ->dehydrateStateUsing(function (News $record, bool $state) {
```

Comment:
> App\Filament\Resources\NewsResource::App\Filament\Resources\{closure}(): Argument #1 ($record) must be of type App\Models\News, null given, called in /var/www/html/vendor/filament/support/src/Concerns/EvaluatesClosures.php on line 35

hitting Post button after filling in a new post.

Reply:
Good catch - fixed in latest.